pkg/agent/openflow/pipeline.go

```diff
@@ -2996,3 +3003,122 @@ func getZoneSrcField(connectUplinkToBridge bool) *binding.RegField {
 	}
 	return nil
 }
+
+func (f *featureMulticluster) l3FwdFlowToLocalGatewayViaTun(
+	localGatewayMAC net.HardwareAddr,
+	peerSubnet net.IPNet,
+	tunnelPeer net.IP,
+	remoteGatewayIP net.IP) []binding.Flow {
+	ipProtocol := getIPProtocol(peerSubnet.IP)
+	cookieID := f.cookieAllocator.Request(f.category).Raw()
+	var flows []binding.Flow
+	flows = append(flows,
+		// This generates the flow to forward cross-cluster request traffic based
+		// on Service ClusterIP range.
+		L3ForwardingTable.ofTable.BuildFlow(priorityNormal).
+			Cookie(cookieID).
+			MatchProtocol(ipProtocol).
+			MatchDstIPNet(peerSubnet).
+			Action().SetSrcMAC(localGatewayMAC).  // Rewrite src MAC to local gateway MAC.
+			Action().SetDstMAC(GlobalVirtualMAC). // Rewrite dst MAC to virtual MAC.
+			Action().SetTunnelDst(tunnelPeer).    // Flow based tunnel. Set tunnel destination.
+			Action().LoadRegMark(ToTunnelRegMark).
+			Action().GotoTable(L3DecTTLTable.GetID()).
+			Done(),
+		// This generates the flow to forward cross-cluster reply traffic based
+		// on Gateway IP.
+		L3ForwardingTable.ofTable.BuildFlow(priorityNormal).
+			Cookie(cookieID).
+			MatchProtocol(ipProtocol).
+			MatchCTStateRpl(true).
+			MatchCTStateTrk(true).
+			MatchDstIP(remoteGatewayIP).
+			Action().SetSrcMAC(localGatewayMAC).
+			Action().SetDstMAC(GlobalVirtualMACForMulticluster).
+			Action().SetTunnelDst(tunnelPeer). // Flow based tunnel. Set tunnel destination.
+			Action().LoadRegMark(ToTunnelRegMark).
+			Action().GotoTable(L3DecTTLTable.GetID()).
+			Done(),
+	)
+	return flows
+}
+
+func (f *featureMulticluster) l3FwdFlowToRemoteViaTun(
+	localGatewayMAC net.HardwareAddr,
+	peerSubnet net.IPNet,
+	tunnelPeer net.IP,
+	remoteGatewayIP net.IP) []binding.Flow {
+	ipProtocol := getIPProtocol(remoteGatewayIP)
+	cookieID := f.cookieAllocator.Request(f.category).Raw()
+	var flows []binding.Flow
+	flows = append(flows,
+		// This generates the flow to forward cross-cluster request traffic based
+		// on Service ClusterIP range.
+		L3ForwardingTable.ofTable.BuildFlow(priorityNormal).
+			Cookie(cookieID).
+			MatchProtocol(ipProtocol).
+			MatchDstIPNet(peerSubnet).
```

Comment:
For packets from regular Node, this will not be true because EndpointDNAT is done in regular Node instead of Gateway Node, this means the traffic is not ct tracked when it's packets from regular Node.